native/acter/src/api/common.rs

```diff
@@ -53,6 +53,20 @@ impl OptionBuffer {
     }
 }
 
+pub struct OptionRsvpStatus {
+    pub(crate) status: Option<RsvpStatus>,
+}
+
+impl OptionRsvpStatus {
+    pub(crate) fn new(status: Option<RsvpStatus>) -> Self {
+        OptionRsvpStatus { status }
+    }
+
+    pub fn status(&self) -> Option<RsvpStatus> {
+        self.status.clone()
+    }
+}
```

Comment:
This is the third of the same sort of code now... Shouldn't we unify these?

```suggestion
pub struct FfiOption<T> (Option<T>)

impl<T> FfiOption<T> where T: Clone{
    pub(crate) fn new(status: Option<T>) -> Self {
        FfiOption(inner)
    }

    pub fn inner(&self) -> Option<T> {
        self.0.clone()
    }
}

type OptionRsvpStatus = FfiOption<RsvpStatus>;
type OptionString = FfiOption<String>;
type OptionBuffer = FfiOption<FfiBuffer<u8>>;
```

?

Reply:
Great idea.
As you noted, `OptionXXXX` code is duplication and I wanted to optimize them, but I didn't know how to implement it.
Now will try your solution.